### PR TITLE
Drive UI display by a IIIF manifest

### PR DESCRIPTION
**Pull request by adamjarling**
## What does this PR do?

In essence, it renders out UI based on data from a Work's IIIF Manifest rather than our API data.  

Since most of our manifests currently referenced in the API Work response contain little to no metadata, this PR implements a helper function which constructs a bare-bones IIIF Presentation 3 manifest.  This manifest object is then delivered to the Work page (at build time), maintaining the static page delivered to end users.

Next, this PR utilizes Nectar to easily display IIIF data on the page.  

image

### CODE REVIEW

File: lib/iiif/manifest-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        label: "Subjects",
+        value: [...work.subject_labels],
+      },
+      {
+        label: "Accession Number",
+        value: work.accession_number,
+      },
+      {
+        label: "Terms of Use",
+        value: work.terms_of_use,
```
Comment (mathewjordan) [inline lib/iiif/manifest-helpers.ts:59]: This is temporary, so no big deal, but I think long term, we'll want this metadata value as an entry in the `requiredStatement` value array and not here. More fitting place for it IIIF wise.

File: lib/iiif/manifest-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      },
+    ]);
+  } catch (err) {
+    throw new Error("Error building manifest locally");
+  }
+  return manifest;
+};
+
+export const getManifest = async (id: string) => {
+  const manifest = await getAPIData<Manifest>({ url: id });
```
Comment (mathewjordan) [inline lib/iiif/manifest-helpers.ts:69]: Simple enough!

File: types/declarations.d.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Declare any modules here which don't have
+ * types exported with their package.  Without
+ * this, TypeScript complains.
+ */
+
+declare module "@samvera/nectar-iiif";
```
Comment (mathewjordan) [inline types/declarations.d.ts:7]: This is nice to have for packages, especially ones out of control. We should probably get on creating types for Nectar.

**Review by mathewjordan — APPROVED** [review]
Looks great.

**Review by mathewjordan — COMMENTED** [review]
It looks like we have some build errors related to Nectar. I think this is related to the `dompurify` package that Nectar uses for sanitizing markup. Apparently that package doesn't play well with node environments and we may want to use https://www.npmjs.com/package/isomorphic-dompurify instead. At any rate, maybe we can pair quickly tomorrow to resolve? Everything else looks good. Seems to be only on build.

File: lib/iiif/manifest-helpers.ts (hunk 1 above)
Comment (adamjarling) [reply lib/iiif/manifest-helpers.ts:59]: Sure.  I was kind of just throwing random metadata into this for now to get some content length happening.  Sounds good on `requiredStatement`.

File: lib/iiif/manifest-helpers.ts (hunk 2 above)
Comment (adamjarling) [reply lib/iiif/manifest-helpers.ts:69]: Yea, maybe there's a better way, but by passing a generic into `getApiData` (ie. `Manifest` here), we can keep `getAPIData`  more simple and put the burden of type checking on the consumers of the function?   Guess that's like TypeScript 101 finally sinking in:)

**Comment by adamjarling** [comment]
@mathewjordan I'm going to merge this branch, as seems unrelated to the Nectar dependency build issue.  And then we make a separate PR for the Nectar upgrade how about?

**Pull request merged**